RecoLocalCalo/HcalRecAlgos/src/parseHFPhase1AlgoDescription.cc

```diff
@@ -30,6 +30,12 @@ parseHFPhase1AlgoDescription(const edm::ParameterSet& ps)
         const bool rejectAllFailures =
             ps.getParameter<bool>("rejectAllFailures");
 
+        float minChargeForUndershoot = -10000.f, minChargeForOvershoot = -10000.f;
+        if (ps.existsAs<double>("minChargeForUndershoot"))
```

Comment:
"parseHFPhase1AlgoDescription" is a factory for classes that inherit from "AbsHFPhase1Algo". It performs similar work (in C++) as the plugin loaders do for module-based architectures (in CMSSW, this work is nicely offloaded to python "import" statements). Right now, we have settled on one particular HF reconstruction algorithm, HFFlexibleTimeCheck, but in the past we have used another (HFSimpleTimeCheck), and it is entirely possible that we will use something else in the future. Right now, parameter descriptions for these algorithms do not exist (we simply require that an algorithm is configured from a ParameterSet). They can be added, but it spells a maintenance nightmare in the future -- as soon as the algorithm needs to be changed (in python), one would have to change "fillDescriptions" (in C++!).